tasks/steps/configure_instance.yml

```diff
@@ -8,17 +8,19 @@
         console_sock: '{{ instance_info.console_sock }}'
         config: '{{ config }}'
         cartridge_defaults: '{{ cartridge_defaults }}'
-      when: not restarted
+      when: (restarted == false) or (restarted is none and not needs_restart)
 
-    - name: 'Check if instance restart is forced or required to apply changes'
-      cartridge_set_needs_restart:
+    - name: 'Check if instance restart is required to update config'
+      cartridge_set_needs_restart_to_configure:
```

Comment:
How does it work when `needs_restart` is `true` after updating package but configuration isn't changed?